src/Components/SelfInterviewThree/utils.ts

```diff
@@ -0,0 +1,28 @@
+import { Question } from "../../Apis/types";
+
+export const sortAlphabetically = (questions: Question[]) => {
+    const sorted = questions.sort((a, b) => {
+        if (a.question > b.question) return -1;
+        if (a.question < b.question) return 1;
+        return 0;
+    });
+
+    return sorted;
+};
+
+export const alternateSort = (questions: Question[]) => {
+    return questions;
+    let result = questions;
+    let j = result.length - 1;
+    let i = 0;
+
+    while (j !== i) {
+        let temp = result[i];
+        result[i] = result[j];
+        result[j] = temp;
+        j -= 1;
+        i += 1;
+    }
+
```

Comment:
The reason that you have an infinite loop is because you have an even number for the array. Since you plus and minus from i and j that means you shift by 2 each time. Let's look at an example:

Odd => [a, b, c]

pass 1 => i = 0; j = 2;
pass 2 => i = 1; j = 1;

Even => [a, b, c, d]

pass 1 => i = 0; j = 3;
pass 2 => i = 1; j = 2;
pass 3 => i = 2; j = 1;
... => infinite loop

As for the solution here it doesn't actually result in the desired solution. If the infinitely loop didn't happen then your solution just reverses the array:

[a, b, c, d, e, f] => [f, e, d, c, b, a]